From a9ca3c37885eafced26b139ff9d0dbfd17811136 Mon Sep 17 00:00:00 2001
From: "Tham, Mun Yew" <mun.yew.tham@intel.com>
Date: Tue, 5 Mar 2024 13:48:14 +0800
Subject: [PATCH] Enable xgmac1 for TSN Config 2 with OOBE2

This patch enables the xgmac1 for Si Dev Kit and
enables the xgmac2 for OOBE2 daughter card.
Xgmac1 will connect with Si Dev Kit on board PHY with address 0.

Signed-off-by: Tham, Mun Yew <mun.yew.tham@intel.com>
---
 .../arm64/boot/dts/intel/socfpga_agilex5.dtsi | 174 +++++++++---------
 .../boot/dts/intel/socfpga_agilex5_socdk.dts  |  20 +-
 2 files changed, 97 insertions(+), 97 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex5.dtsi b/arch/arm64/boot/dts/intel/socfpga_agilex5.dtsi
index 3c02717f30ef..2c62ded572aa 100644
--- a/arch/arm64/boot/dts/intel/socfpga_agilex5.dtsi
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex5.dtsi
@@ -231,28 +231,28 @@
 			#clock-cells = <1>;
 		};
 
-		gmac0: ethernet@10810000 {
+gmac2: ethernet@10830000 {
 			compatible = "altr,socfpga-stmmac-a10-s10",
 				     "snps,dwxgmac-2.10",
 				     "snps,dwxgmac";
-			reg = <0x10810000 0x3500>;
-			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 199 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 200 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 204 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 205 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 206 IRQ_TYPE_LEVEL_HIGH>;
+			reg = <0x10830000 0x3500>;
+			interrupts = <GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 225 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 227 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 229 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 230 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 231 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 232 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 233 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 234 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 235 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 236 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 237 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 238 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 239 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 240 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "macirq",
 					  "macirq_tx0",
 					  "macirq_tx1",
@@ -270,9 +270,9 @@
 					  "macirq_rx5",
 					  "macirq_rx6",
 					  "macirq_rx7";
-			resets = <&rst EMAC0_RESET>, <&rst EMAC0_OCP_RESET>;
+			resets = <&rst EMAC2_RESET>, <&rst EMAC2_OCP_RESET>;
 			reset-names = "stmmaceth", "stmmaceth-ocp";
-			clocks = <&clkmgr AGILEX5_EMAC0_CLK>,
+			clocks = <&clkmgr AGILEX5_EMAC2_CLK>,
 				 <&clkmgr AGILEX5_EMAC_PTP_CLK>;
 			clock-names = "stmmaceth", "ptp_ref";
 			mac-address = [00 00 00 00 00 00];
@@ -280,27 +280,27 @@
 			rx-fifo-depth = <16384>;
 			snps,multicast-filter-bins = <64>;
 			snps,perfect-filter-entries = <64>;
-			snps,axi-config = <&stmmac_axi_emac0_setup>;
-			snps,mtl-rx-config = <&mtl_rx_emac0_setup>;
-			snps,mtl-tx-config = <&mtl_tx_emac0_setup>;
+			snps,axi-config = <&stmmac_axi_emac2_setup>;
+			snps,mtl-rx-config = <&mtl_rx_emac2_setup>;
+			snps,mtl-tx-config = <&mtl_tx_emac2_setup>;
 			snps,pbl = <32>;
 			snps,pblx8;
 			snps,multi-irq-en;
 			snps,tso;
 			snps,rx-vlan-offload;
-			altr,sysmgr-syscon = <&sysmgr 0x44 0>;
+			altr,sysmgr-syscon = <&sysmgr 0x4c 0>;
 			altr,smtg-hub;
-			iommus = <&smmu 1>;
+			iommus = <&smmu 3>;
 
 			status = "disabled";
 
-			stmmac_axi_emac0_setup: stmmac-axi-config {
+			stmmac_axi_emac2_setup: stmmac-axi-config {
 				snps,wr_osr_lmt = <31>;
 				snps,rd_osr_lmt = <31>;
 				snps,blen = <0 0 0 32 16 8 4>;
 			};
 
-			mtl_rx_emac0_setup: rx-queues-config {
+			mtl_rx_emac2_setup: rx-queues-config {
 				snps,rx-queues-to-use = <8>;
 				snps,rx-sched-sp;
 				queue0 {
@@ -337,7 +337,7 @@
 				};
 			};
 
-			mtl_tx_emac0_setup: tx-queues-config {
+			mtl_tx_emac2_setup: tx-queues-config {
 				snps,tx-queues-to-use = <8>;
 				snps,tx-queues-with-coe = <2>;
 				snps,tx-sched-wrr;
@@ -378,28 +378,28 @@
 			};
 		};
 
-		gmac1: ethernet@10820000 {
+		gmac0: ethernet@10810000 {
 			compatible = "altr,socfpga-stmmac-a10-s10",
 				     "snps,dwxgmac-2.10",
 				     "snps,dwxgmac";
-			reg = <0x10820000 0x3500>;
-			interrupts = <GIC_SPI 207 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 209 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 212 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 213 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 214 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 215 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 216 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 217 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 218 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 219 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 221 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 222 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
+			reg = <0x10810000 0x3500>;
+			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 199 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 200 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 204 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 205 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 206 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "macirq",
 					  "macirq_tx0",
 					  "macirq_tx1",
@@ -417,9 +417,9 @@
 					  "macirq_rx5",
 					  "macirq_rx6",
 					  "macirq_rx7";
-			resets = <&rst EMAC1_RESET>, <&rst EMAC1_OCP_RESET>;
+			resets = <&rst EMAC0_RESET>, <&rst EMAC0_OCP_RESET>;
 			reset-names = "stmmaceth", "stmmaceth-ocp";
-			clocks = <&clkmgr AGILEX5_EMAC1_CLK>,
+			clocks = <&clkmgr AGILEX5_EMAC0_CLK>,
 				 <&clkmgr AGILEX5_EMAC_PTP_CLK>;
 			clock-names = "stmmaceth", "ptp_ref";
 			mac-address = [00 00 00 00 00 00];
@@ -427,27 +427,27 @@
 			rx-fifo-depth = <16384>;
 			snps,multicast-filter-bins = <64>;
 			snps,perfect-filter-entries = <64>;
-			snps,axi-config = <&stmmac_axi_emac1_setup>;
-			snps,mtl-rx-config = <&mtl_rx_emac1_setup>;
-			snps,mtl-tx-config = <&mtl_tx_emac1_setup>;
+			snps,axi-config = <&stmmac_axi_emac0_setup>;
+			snps,mtl-rx-config = <&mtl_rx_emac0_setup>;
+			snps,mtl-tx-config = <&mtl_tx_emac0_setup>;
 			snps,pbl = <32>;
 			snps,pblx8;
 			snps,multi-irq-en;
 			snps,tso;
 			snps,rx-vlan-offload;
-			altr,sysmgr-syscon = <&sysmgr 0x48 0>;
+			altr,sysmgr-syscon = <&sysmgr 0x44 0>;
 			altr,smtg-hub;
-			iommus = <&smmu 2>;
+			iommus = <&smmu 1>;
 
 			status = "disabled";
 
-			stmmac_axi_emac1_setup: stmmac-axi-config {
+			stmmac_axi_emac0_setup: stmmac-axi-config {
 				snps,wr_osr_lmt = <31>;
 				snps,rd_osr_lmt = <31>;
 				snps,blen = <0 0 0 32 16 8 4>;
 			};
 
-			mtl_rx_emac1_setup: rx-queues-config {
+			mtl_rx_emac0_setup: rx-queues-config {
 				snps,rx-queues-to-use = <8>;
 				snps,rx-sched-sp;
 				queue0 {
@@ -484,7 +484,7 @@
 				};
 			};
 
-			mtl_tx_emac1_setup: tx-queues-config {
+			mtl_tx_emac0_setup: tx-queues-config {
 				snps,tx-queues-to-use = <8>;
 				snps,tx-queues-with-coe = <2>;
 				snps,tx-sched-wrr;
@@ -525,28 +525,28 @@
 			};
 		};
 
-		gmac2: ethernet@10830000 {
+		gmac1: ethernet@10820000 {
 			compatible = "altr,socfpga-stmmac-a10-s10",
 				     "snps,dwxgmac-2.10",
 				     "snps,dwxgmac";
-			reg = <0x10830000 0x3500>;
-			interrupts = <GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 225 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 227 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 229 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 230 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 231 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 232 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 233 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 234 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 235 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 236 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 237 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 238 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 239 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 240 IRQ_TYPE_LEVEL_HIGH>;
+			reg = <0x10820000 0x3500>;
+			interrupts = <GIC_SPI 207 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 209 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 212 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 213 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 214 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 215 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 216 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 217 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 218 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 219 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 221 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 222 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "macirq",
 					  "macirq_tx0",
 					  "macirq_tx1",
@@ -564,9 +564,9 @@
 					  "macirq_rx5",
 					  "macirq_rx6",
 					  "macirq_rx7";
-			resets = <&rst EMAC2_RESET>, <&rst EMAC2_OCP_RESET>;
+			resets = <&rst EMAC1_RESET>, <&rst EMAC1_OCP_RESET>;
 			reset-names = "stmmaceth", "stmmaceth-ocp";
-			clocks = <&clkmgr AGILEX5_EMAC2_CLK>,
+			clocks = <&clkmgr AGILEX5_EMAC1_CLK>,
 				 <&clkmgr AGILEX5_EMAC_PTP_CLK>;
 			clock-names = "stmmaceth", "ptp_ref";
 			mac-address = [00 00 00 00 00 00];
@@ -574,27 +574,27 @@
 			rx-fifo-depth = <16384>;
 			snps,multicast-filter-bins = <64>;
 			snps,perfect-filter-entries = <64>;
-			snps,axi-config = <&stmmac_axi_emac2_setup>;
-			snps,mtl-rx-config = <&mtl_rx_emac2_setup>;
-			snps,mtl-tx-config = <&mtl_tx_emac2_setup>;
+			snps,axi-config = <&stmmac_axi_emac1_setup>;
+			snps,mtl-rx-config = <&mtl_rx_emac1_setup>;
+			snps,mtl-tx-config = <&mtl_tx_emac1_setup>;
 			snps,pbl = <32>;
 			snps,pblx8;
 			snps,multi-irq-en;
 			snps,tso;
 			snps,rx-vlan-offload;
-			altr,sysmgr-syscon = <&sysmgr 0x4c 0>;
+			altr,sysmgr-syscon = <&sysmgr 0x48 0>;
 			altr,smtg-hub;
-			iommus = <&smmu 3>;
+			iommus = <&smmu 2>;
 
 			status = "disabled";
 
-			stmmac_axi_emac2_setup: stmmac-axi-config {
+			stmmac_axi_emac1_setup: stmmac-axi-config {
 				snps,wr_osr_lmt = <31>;
 				snps,rd_osr_lmt = <31>;
 				snps,blen = <0 0 0 32 16 8 4>;
 			};
 
-			mtl_rx_emac2_setup: rx-queues-config {
+			mtl_rx_emac1_setup: rx-queues-config {
 				snps,rx-queues-to-use = <8>;
 				snps,rx-sched-sp;
 				queue0 {
@@ -631,7 +631,7 @@
 				};
 			};
 
-			mtl_tx_emac2_setup: tx-queues-config {
+			mtl_tx_emac1_setup: tx-queues-config {
 				snps,tx-queues-to-use = <8>;
 				snps,tx-queues-with-coe = <2>;
 				snps,tx-sched-wrr;
-- 
2.26.2

